libs/parallelism/algorithms/include/hpx/parallel/algorithms/ends_with.hpp

```diff
@@ -0,0 +1,301 @@
+
+                if (drop < 0)
+                    return false;
+
+                return hpx::parallel::v1::detail::equal_binary().call(
+                    hpx::execution::seq, std::next(std::move(first1), drop),
+                    std::move(last1), std::move(first2), std::move(last2),
+                    std::forward<Pred>(pred), std::forward<Proj1>(proj1),
+                    std::forward<Proj2>(proj2));
+            }
+
+            template <typename ExPolicy, typename FwdIter1, typename Sent1,
+                typename FwdIter2, typename Sent2, typename Pred,
+                typename Proj1, typename Proj2>
+            static typename util::detail::algorithm_result<ExPolicy, bool>::type
+            parallel(ExPolicy&& policy, FwdIter1 first1, Sent1 last1,
+                FwdIter2 first2, Sent2 last2, Pred&& pred, Proj1&& proj1,
+                Proj2&& proj2)
+            {
+                const auto drop = detail::distance(first1, last1) -
+                    detail::distance(first2, last2);
+
+                if (drop < 0)
+                {
+                    return util::detail::algorithm_result<ExPolicy, bool>::get(
+                        false);
+                }
+
+                return hpx::parallel::v1::detail::equal_binary().call(
+                    std::forward<ExPolicy>(policy),
+                    std::next(std::move(first1), drop), std::move(last1),
+                    std::move(first2), std::move(last2),
+                    std::forward<Pred>(pred), std::forward<Proj1>(proj1),
+                    std::forward<Proj2>(proj2));
+            }
+        };
+        /// \endcond
+    }    // namespace detail
+}}}      // namespace hpx::parallel::v1
+
+namespace hpx {
+
+    ///////////////////////////////////////////////////////////////////////////
+    // DPO for hpx::ends_with
+    HPX_INLINE_CONSTEXPR_VARIABLE struct ends_with_t final
+      : hpx::functional::tag_fallback<ends_with_t>
+    {
+    private:
+        // clang-format off
+        template <typename InIter1, typename InIter2,
+            typename Pred = hpx::parallel::v1::detail::equal_to,
+            typename Proj1 = parallel::util::projection_identity,
+            typename Proj2 = parallel::util::projection_identity,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<InIter1>::value &&
+                hpx::traits::is_iterator<InIter2>::value &&
+                hpx::parallel::traits::is_indirect_callable<
+                    hpx::execution::sequenced_policy, Pred,
+                    hpx::parallel::traits::projected<Proj1, InIter1>,
+                    hpx::parallel::traits::projected<Proj2, InIter2>
+                >::value
+            )>
+        // clang-format on
+        friend bool tag_fallback_dispatch(hpx::ends_with_t, InIter1 first1,
+            InIter1 last1, InIter2 first2, InIter2 last2, Pred&& pred = Pred(),
+            Proj1&& proj1 = Proj1(), Proj2&& proj2 = Proj2())
```

Comment:
The algorithms in `std` namespace usually don't have projections, only the ones in `std::ranges` do have those.